### ISSUE TITLE
Rate limit possible?

### ISSUE THREAD

**Issue by giddyhup** [shmuelzon/esp32-ble2mqtt#166]
**Is your feature request related to a problem? Please describe.**

I use the solution to forward data from a treadmill. The treadmill's BT MAC is the only whitelisted device. Every once in a while the treadmill shuts down (rapidly slows down) mid-run. This only happens when BLE2MQTT is being used. 

A passively connected client should not cause this, IMHO, yet it happens and it is annoying. Regrettably, I cannot modify the treadmill to behave better. 

**Describe the solution you'd like**

To me the message rate seems rather high and perhaps the treadmill gets overloaded with queries. Perhaps a rate limit for messages could help. I would try if setting such a limit to 1 message/s would help in my case.

**Describe alternatives you've considered**

Not using BLE2MQTT.

**Additional context**

I do not understand the communication so I do not know if this would help but I ask here and see if there something that can be done on the receiving end.

**Comment by shmuelzon** [comment]
@giddyhup, I agree it doesn't sound smart for a treadmill to rapidly slow down when it disconnects from Bluetooth...
Anyway, the way this project is set up, is that we connect to a BLE peripheral, query all of the available characteristics, read their values, register for notifications, and then wait silently.
So, apart for the BLE protocol messages meant to keep the connection alive (which we don't have much control over), we don't actively send anything to the BLE peripheral but rather wait for it to tell us what changed.

If it constantly sends updates on characteristics that aren't important to you, we can blacklist them so only the relevant information is sent. If the connection just times-out and disconnects, there may be other factors that we can influence.
But, before all that, it would be great to understand what happens on the ESP32 side when the treadmill shuts down.

If you can, I suggest capturing logs from the application and, if possible, mark at which point the treadmill stopped. Once we have that, we can continue.

**Comment by giddyhup** [comment]
@shmuelzon thanks for your reply. It is as I expected, we do not have much control. I'll see if I can capture when I have a setup for serial logging again.

**Comment by shmuelzon** [comment]
Assuming Wi-Fi isn't an issue, you can always use the remote logging mechanism to capture the logs. It might not contain everything but it could help, in the meantime...
